eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -615,9 +615,10 @@ case class Commitment(fundingTxIndex: Long,
     // ourCommit.index + 2 -> which is about to become our next revocation hash
     // we will reply to this sig with our old revocation hash preimage (at index) and our next revocation hash (at index + 1)
     // and will increment our index
+    val localCommitIndex = if (isSplice) localCommit.index else localCommit.index + 1
```

Comment:
This is a very leaky abstraction, we can do better. Since splices don't actually have a `Commitment` object yet on which to call `receiveCommit`, it makes more sense to move most of the contents of `receiveCommit` to an outside helper function.

Reply:
Thanks - this is a much cleaner solution. Pulled in [08d7f35](https://github.com/ACINQ/eclair/pull/2720/commits/08d7f352e501d46bed498d6ed468ab838972f2db) unchanged.